src/shared/ui/subtitle/subtitle.tsx

```diff
@@ -1,10 +1,55 @@
-import './subtitle.scss';
+import { HTMLAttributes } from 'react';
+import { type VariantProps, cva } from 'class-variance-authority';
+import classNames from 'classnames/bind';
 
-interface SubtitleProps {
-  text: string;
-  type?: string;
-}
+import styles from './subtitle.module.scss';
 
-export const Subtitle = ({ text, type = '' }: SubtitleProps) => (
-  <div className={`subtitle ${type}`}>{text}</div>
-);
+type SubtitleProps = HTMLAttributes<HTMLHeadingElement> & VariantProps<typeof subtitleVariants>;
+
+export const cx = classNames.bind(styles);
+
+const subtitleVariants = cva(cx('subtitle'), {
+  variants: {
+    fontSize: {
+      medium: cx('medium'),
+      large: cx('large'),
+    },
+    color: {
+      gray: cx('gray-600'),
+      black: cx('black'),
+    },
+    marginSize: {
```

Comment:
We decided not to use `margin` prop in `shared` components and replacing them with `padding`. But I think their size should be determined by the `size` property